cpp/dolfinx/fem/interpolate.h

```diff
@@ -396,6 +397,111 @@ void interpolate(Function<T>& u, const Function<T>& v)
           u_array[u_bs * dofs_u[i] + k] = u_local[u_bs * i + k];
     }
   }
+  else
+  {
+    // get points from v's element
+    // for each cell:
+    //   push points to cell
+    //   evaluate u at points
+    //   interpolate into v's element
+    //   put values into correct entries in dofmap
+
+    //  --- Different elements, different map
+    xtl::span<const std::uint32_t> cell_info;
+    if (element_to->needs_dof_transformations()
+        or element_from->needs_dof_transformations())
+    {
+      mesh->topology_mutable().create_entity_permutations();
+      cell_info = xtl::span(mesh->topology().get_cell_permutation_info());
+    }
+
+    // Get dofmaps
+    const auto dofmap_u = u.function_space()->dofmap();
+    const auto dofmap_v = v.function_space()->dofmap();
+
+    const xt::xtensor<double, 2> points = element_to->interpolation_points();
+
+    // Get block sizes and dof transformation operators
+    const int u_bs = element_to->block_size();
+    const auto apply_dof_transformation
+        = element_from->get_dof_transformation_function<T>(false, true, false);
+    const auto apply_inverse_dof_transform
+        = element_to->get_dof_transformation_function<T>(true, true, false);
+
+    const fem::CoordinateElement& cmap = mesh->geometry().cmap();
+
+    // Get geometry data
+    const graph::AdjacencyList<std::int32_t>& x_dofmap
+        = mesh->geometry().dofmap();
+    // FIXME: Add proper interface for num coordinate dofs
+    const std::size_t num_dofs_g = x_dofmap.num_links(0);
+    const xt::xtensor<double, 2>& x_g = mesh->geometry().x();
+
+    // Create working arrays
+    std::vector<T> u_local(element_to->space_dimension());
+    xt::xtensor<T, 3> v_values({points.shape(0), 1, element_to->value_size()});
+    xt::xtensor<T, 3> v_mapped_values(
+        {points.shape(0), 1, element_to->value_size()});
+    std::vector<int> cells(points.shape(0));
+    std::vector<int> cell(1);
+    xt::xtensor<double, 4> phi(cmap.tabulate_shape(1, points.shape(0)));
+    xt::xtensor<double, 2> dphi;
+    xt::xtensor<double, 2> coordinate_dofs({num_dofs_g, gdim});
+    xt::xtensor<double, 3> J({points.shape(0), gdim, tdim});
+    xt::xtensor<double, 3> K({points.shape(0), tdim, gdim});
+    const std::array<std::size_t, 1> sh = {points.shape(0)};
+    xt::xtensor<double, 1> detJ(sh);
+
+    // Iterate over mesh and interpolate on each cell
+    const int num_cells = map->size_local() + map->num_ghosts();
+    for (int c = 0; c < num_cells; ++c)
+    {
+      // Get cell geometry (coordinate dofs)
+      auto x_dofs = x_dofmap.links(c);
+      for (std::size_t i = 0; i < num_dofs_g; ++i)
+        for (std::size_t j = 0; j < gdim; ++j)
+          coordinate_dofs(i, j) = x_g(x_dofs[i], j);
+
+      // Evaluate v at interpolation points
+      cell[0] = c;
+      for (std::size_t i = 0; i < cells.size(); ++i)
+        cells[i] = c;
+      const xt::xtensor<double, 2> X
+          = interpolation_coords(*element_to, *mesh, cell);
```

Comment:
Could we push points to the cell instead of using `interpolation_coords` here so that we don't need to transpose later?

Reply:
This has been resolved by adding a `transpose` argument to `interpolation_coords`.